src/notifications/screens/notifications.screen.js

```diff
@@ -233,115 +309,104 @@ class Notifications extends Component {
     }
   }
 
-  notifications() {
-    const { unread, participating, all } = this.props;
-    const { type } = this.state;
+  navigateToThread(notification) {
+    const { markAsRead, navigation } = this.props;
 
-    switch (type) {
-      case 0:
-        return unread;
-      case 1:
-        return participating;
-      case 2:
-        return all;
-      default:
-        return null;
-    }
+    markAsRead(notification.id);
+    navigation.navigate('Issue', {
+      issueURL: notification.subject.url.replace('pulls', 'issues'),
+      isPR: notification.subject.type === 'PullRequest',
+      language: this.props.language,
+    });
   }
 
-  isLoading() {
-    const {
-      unread,
-      participating,
-      all,
-      isPendingUnread,
-      isPendingParticipating,
-      isPendingAll,
-    } = this.props;
-    const { type } = this.state;
-
-    switch (type) {
-      case 0:
-        return unread && isPendingUnread;
-      case 1:
-        return participating && isPendingParticipating;
-      case 2:
-        return all && isPendingAll;
-      default:
-        return null;
-    }
-  }
+  navigateToRepo = fullName => {
+    const { navigation } = this.props;
 
-  keyExtractor = (item, index) => {
-    return index;
+    navigation.navigate('Repository', {
+      repositoryUrl: `${v3.root}/repos/${fullName}`,
+    });
   };
 
   renderItem = ({ item }) => {
-    const { markAsRead, markRepoAsRead } = this.props;
+    const isFirst = this.getSortedRepos().indexOf(item) === 0;
+
+    const {
+      markAsRead,
+      markRepoAsRead,
+      markAllNotificationsAsRead,
+    } = this.props;
     const notifications = this.notifications().filter(
       notification => notification.repository.full_name === item
     );
 
     return (
-      <Card containerStyle={styles.repositoryContainer}>
-        <View style={styles.headerContainer}>
-          <Image
-            style={styles.repositoryOwnerAvatar}
-            source={{
-              uri: this.getImage(item),
+      <View>
+        {isFirst &&
+          <Button
+            icon={{
+              name: 'check',
+              size: 20,
+              type: 'octicon',
+              color: colors.black,
             }}
-          />
-
-          <Text
-            style={styles.repositoryTitle}
-            onPress={() => this.navigateToRepo(item)}
-          >
-            {item}
-          </Text>
-
-          <TouchableOpacity
-            style={styles.markAsReadIconRepo}
-            onPress={() => markRepoAsRead(item)}
-          >
-            <Icon
-              color={colors.greyDark}
-              size={28}
-              name="check"
-              type="octicon"
+            title={translate('notifications.main.markAllAsRead')}
+            buttonStyle={styles.markAllAsReadButton}
+            color={colors.black}
+            backgroundColor={colors.white}
+            textStyle={styles.buttonGroupText}
+            onPress={() => markAllNotificationsAsRead()}
```

Comment:
Does this button show when `unread`, `pending` _and_ `all` tabs are selected? Not sure if that's the best way of doing things. Do you think it will make more sense to have it only show only when `all` is selected (and then maybe having buttons that show `Mark all unread notifications as read` and `Mark all participating notifications as read` for the other 2 tabs?

Reply:
Even if you think we don't need the other two buttons in this PR, I think having the Mark ALL button in each of the tabs can be kind of misleading

Reply:
Currently, on notifications [page](https://github.com/notifications) this button is displayed everywhere, and it always marks notifications as read _all_ notifications, i.e. there is no way to mark as read only _participating_ notifications.

Let's display this button only on the first tab (_unread_)?

Reply:
Ah that's a very good point 🤔 

Hmmmmmm, I think I'm okay with having it display only on the first tab in the case. If I'm not mistaken, all _participating_ notifications are _unread_ notifications anyway so this should be okay 👍